### PR TITLE
Fortran package manager in programming introduction

### LINKED ISSUE #18: Use Fortran Package Manager in the Fortran Introduction
One of the main issues observed in the past is the steep learning curve when going from the plain Fortran compiler to a build system like `make`.

A more consistent interface to Fortran would make the transition from the examples to the final programming project smoother in my opinion. While using a build system from the beginning of the course seemed like an unnecessary complication with `make` (or `cmake`, `meson`, whatever), the Fortran Package Manager ([`fpm`](https://github.com/fortran-lang/fpm)) instead is offering an easy and self-explanatory user interface required for such an approach.

This issue should track the progress and possibility to use `fpm` for this course.

### PR DESCRIPTION
**Pull request by awvwgk**
This patch introduces the Fortran package manager (fpm) for the programming part. While still in alpha stage it is already incredible useful and will hopefully improve the overall learning experience in the programming course. We will show information for both fpm and gfortran in the programming course switchable by tabs to allow completing the course with and without fpm.

The only and main caveat: the installation will become more complicated.

Preview at https://qc2-teaching.readthedocs.io/en/fpm

Closes #18

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -97,91 +153,91 @@ You have now successfully installed a Linux console on your Windows computer and
...
-   point for your SCF program later.
+   Alternatively, you can create a separate environment for fpm with
+
+   .. code-block:: bash
+
+      mamba create -n fpm fpm
+      conda activate fpm
+
+   When creating a separate environment, you will always have to activate it after
+   when starting a new shell, but it allows you more fine grained control over
```
Comment (MtoLStoN) [inline doc/install.rst:239]: Wording
```suggestion
   starting a new shell, but it allows you more fine grained control over
```

File: doc/programming.rst [hunk 2]
```diff
@@ -146,6 +146,73 @@ The double quotes enclosing the sentence make the program recognize that
...
+The Fortran package manager (fpm) is a recent advancement introduced by the
+Fortran community. It helps to abstract common tasks when developing Fortran
+applications like building and running executables or creating new projects.
+
+.. warning::
+
+   The Fortran package manager is still a relatively new project.
+   Using new and recent software always has some risks of running into
+   unexpected issues. We have carefully evaluated fpm and the advantage
+   of the simple user interface outweigh potential issues.
```
Comment (MtoLStoN) [inline doc/programming.rst:161]: Grammar
```suggestion
   of the simple user interface outweighs potential issues.
```

File: doc/programming.rst [hunk 3]
```diff
@@ -146,6 +146,73 @@ The double quotes enclosing the sentence make the program recognize that
...
+   By default fpm enables compile time checks and run time checks that
+   are absent in the plain compiler invocation. Those can help you catch
+   and avoid common errors when developing in Fortran.
+
+   To read more on the capabilities of fpm check the help page with
+
+   .. code-block::
+
+      fpm help
+
```
Comment (MtoLStoN) [inline doc/programming.rst:214]: Maybe it would be beneficial for the students to use fpm build instead of fpm run?

**Review by MtoLStoN — CHANGES_REQUESTED** [review]
Nicely done, marked a few things, but looks good to me otherwise.

File: doc/programming.rst (hunk 3 above)
Comment (MtoLStoN) [reply doc/programming.rst:214]: But is it possible to use fpm run with a command-line syntax? I'm not sure, how this would work?

Comment (awvwgk) [reply doc/programming.rst:214]: Oh, you can pass-through arguments by the usual Unix like shell syntax:
```
fpm run -- --arg
```
Everything after ``--`` goes unmodified as argument to the executable.

Comment (MtoLStoN) [reply doc/programming.rst:214]: Oh, great - I did not know that

Comment (awvwgk) [reply doc/programming.rst:214]: But you are right, I should highlight this feature since it will be used in the SCF part.

**Review by MtoLStoN — APPROVED** [review]
Looks good now.

**Pull request merged**